.github/workflows/check_deps_integrity.yaml

```
@@ -14,32 +14,30 @@ jobs:
   check_deps_integrity:
     runs-on: ${{ endsWith(github.repository, '/emqx') && 'ubuntu-22.04' || fromJSON('["self-hosted","ephemeral","linux","x64"]') }}
     container: ${{ inputs.builder }}
+    env:
+      MIX_ENV: ${{ matrix.profile }}
+      PROFILE: ${{ matrix.profile }}
+    strategy:
+      matrix:
+        profile:
+          - emqx-enterprise
     steps:
       - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
       - run: git config --global --add safe.directory "$GITHUB_WORKSPACE"
       - run: make ensure-rebar3
       - run: ./scripts/check-deps-integrity.escript
       - name: Setup mix
-        env:
-          MIX_ENV: emqx-enterprise
-          PROFILE: emqx-enterprise
         run: |
           mix local.hex --force
           mix local.rebar --force
           mix deps.get
       - run: ./scripts/check-elixir-deps-discrepancies.exs
```

Just a nano suggestion that might help debug future issues, since we're touching here:

```suggestion
      - name: print mix dependency tree
        run: mix deps.tree
      - run: ./scripts/check-elixir-deps-discrepancies.exs
```